great_tables/_tab_create_modify.py

```diff
@@ -118,6 +119,34 @@ def tab_style(
         locations = [locations]
 
     new_data = self
+
+    # Intercept `font` in CellStyleText to capture Google Fonts and:
+    # 1. transform dictionary to string (with Google Font name)
+    # 2. add Google Font import statement via tab_options(table_additional_css)
+    if any(isinstance(s, CellStyle) for s in style):
+
+        for s in style:
+            if (
+                isinstance(s, CellStyle)
+                and hasattr(s, "font")
+                and s.font is not None
+                and isinstance(s.font, GoogleFont)
+            ):
+                # Obtain font name and import statement as local variables
+                font_name = s.font.get_font_name()
+                font_import_stmt = s.font.make_import_stmt()
+
+                # Replace GoogleFont class with font name
+                s.font = font_name
+
+                # Append the import statement to the `table_additional_css` list
+                existing_additional_css = self._options.table_additional_css.value + [
+                    font_import_stmt
+                ]
+
+                # Add revised CSS list via the `tab_options()` method
+                new_data = new_data.tab_options(table_additional_css=existing_additional_css)
```

Comment:
For pairing Thursday, let's go over alternative paths this could go through. Currently, CellStyle as an interface has a `_to_html_style()` method, that gets used on render.

Here, it looks like the new behavior to support is something like this:

* A certain type of style requires setting some initial table-level CSS (once for the importing of fonts)

Currently, the path used for this is:

* in tab_style, detect a particular CellStyleText (e.g. with a google font)
* create import statement for font
* set initial table css for importing font in the additional_css option
* additional css gets included on render

However, this means that if you use tab_style multiple times with google font, etc.., the additional css gets appended multiple times. Moreover, if the person modifies additional_css, they can remove the necessary import.

A second issue is that this exact code is pasted into the `opt_table_font()` method. We likely don't want to put custom handling for specific CellStyle types (with specific attributes) into these methods.

Repeating code twice isn't necessarily an issue though, so it could be that we map out paths for these kinds of activities, merge, and punt the work to a new issue

Reply:
Sounds good!